sdk/core/azure-core-http-jdk11-httpclient/src/main/java/com/azure/core/http/jdk/httpclient/JdkAsyncHttpClient.java

```diff
@@ -0,0 +1,197 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.http.jdk.httpclient;
+
+import com.azure.core.http.HttpClient;
+import com.azure.core.http.HttpHeader;
+import com.azure.core.http.HttpHeaders;
+import com.azure.core.http.HttpRequest;
+import com.azure.core.http.HttpResponse;
+import com.azure.core.util.CoreUtils;
+import com.azure.core.util.FluxUtil;
+import reactor.adapter.JdkFlowAdapter;
+import reactor.core.Exceptions;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import java.net.URISyntaxException;
+import java.nio.ByteBuffer;
+import java.nio.charset.Charset;
+import java.util.Collections;
+import java.util.List;
+
+import java.net.http.HttpRequest.BodyPublisher;
+import java.util.Set;
+import java.util.TreeSet;
+import java.util.concurrent.Flow;
+
+import static java.net.http.HttpResponse.BodyHandlers.*;
+import static java.net.http.HttpRequest.BodyPublishers.*;
+
+/**
+ * HttpClient implementation for the JDK 11 HttpClient.
+ */
+class JdkAsyncHttpClient implements HttpClient {
+    private final java.net.http.HttpClient jdk11HttpClient;
+
+    private static final Set<String> JDK11_RESTRICTED_HEADERS;
+    static {
+        TreeSet<String> treeSet = new TreeSet<>(String.CASE_INSENSITIVE_ORDER);
```

Comment:
This should be more flexible in its configuration, if someone is using Java 12+ and set the system property to allow restricted headers we should allow these headers.

Reply:
I added this initially to avoid JDK11 HttpClient native builder from throwing an exception. Thinking what would be the best approach here:

Maybe we can copy over the same jdk11+ logic around the header whitelisting here or if the exception thrown by the JDK11+ HttpClient native builder has a special error code for disallowed headers then we can catch, log and continue.

Reply:
Relaxed validations and added necessary log messages as agreed offline.